app/code/Magento/Catalog/Model/Product/Image.php

```diff
@@ -278,7 +278,7 @@ public function getQuality()
      */
     public function setKeepAspectRatio($keep)
     {
-        $this->_keepAspectRatio = (bool)$keep;
+        $this->_keepAspectRatio = filter_var($keep, FILTER_VALIDATE_BOOLEAN);
```

Comment:
Would something like
```
$this->_keepAspectRatio = $keep && $keep !== 'false';
```
do the trick and make an intention clearer for somebody who read the code? ;)